src/facade/dragonfly_connection.cc

```diff
@@ -729,6 +756,10 @@ void Connection::DispatchFiber(util::FiberSocketBase* peer) {
     RequestPtr req{std::move(dispatch_q_.front())};
     dispatch_q_.pop_front();
     std::visit(dispatch_op, req->payload);
+
+    // Do not cache more than K items.
+    if (free_req_pool_.size() < 16)
+      free_req_pool_.push_back(std::move(req));
```

Comment:
So you collect them of all kinds of types, but use it only to avoid allocations for requests with a PipelineMsg payload

Reply:
yeah, I did not implement all the cases but as long as it does not affect correctness I am fine with solving a narrow use-case.